tests/test_enterprise/test_utils.py

```diff
@@ -238,3 +250,85 @@ def test_enroll_pending_licensed_users_in_courses_succeeds(self):
         self.assertFalse(result['failures'])
+
+    def setup_notification_test_data(self):
+        """
+        Creates data needed for testing serialization of email notifications data
+        """
+        ent_customer = factories.EnterpriseCustomerFactory(
+            uuid=FAKE_UUIDS[0],
+            name="test_enterprise"
+        )
+        user = factories.UserFactory()
+        pending_user = factories.PendingEnterpriseCustomerUserFactory()
+        users = [user, pending_user]
+
+        course_id = 'course-v1:edx+123+T2021'
+        course_details = {'title': 'a_course', 'start': '2021-01-01T00:10:10', 'course': 'edx+123'}
+        return ent_customer, users, course_id, course_details
+
+    @mock.patch("enterprise.utils.get_configuration_value_for_site")
+    def test_serialize_notification_content_self_enroll(self, mock_get_config_value_for_site):
+        mock_get_config_value_for_site.return_value = LMS_BASE_URL
+        ent_customer, users, course_id, course_details = self.setup_notification_test_data()
+
+        email_items = serialize_notification_content(ent_customer, course_details, course_id, users)
+
+        def expected_email_item(user):
+            course_path = '/courses/{course_id}/course'.format(course_id=course_id)
+            login_or_register = 'register' if is_pending_user(user) else 'login'
+            return {
+                "user": model_to_dict(user, fields=['first_name', 'username', 'user_email', 'email']),
+                "enrolled_in": {
+                    'name': course_details.get('title'),
+                    'url': "{lms_base_url}/{login_or_register}?next={course_path}".format(
+                        lms_base_url=LMS_BASE_URL,
+                        course_path=urlquote("{}?{}".format(course_path, urlencode([]))),
+                        login_or_register=login_or_register,
+                    ),
+                    'type': 'course',
+                    'start': parse_lms_api_datetime(course_details.get('start'))
+                },
+                "dashboard_url": None,
+                "enterprise_customer_uuid": ent_customer.uuid,
+                "admin_enrollment": False,
+            }
+
+        expected_email_items = [expected_email_item(user) for user in users]
+        assert email_items == expected_email_items
+
+    @mock.patch("enterprise.utils.get_configuration_value_for_site")
+    def test_serialize_notification_content_admin_enroll(self, mock_get_config_value_for_site):
+        mock_get_config_value_for_site.return_value = LMS_BASE_URL
+        ent_customer, users, course_id, course_details = self.setup_notification_test_data()
+
+        email_items = serialize_notification_content(
+            ent_customer,
+            course_details,
+            course_id,
+            users,
+            admin_enrollment=True,
+        )
+
+        def expected_email_item(user):
+            course_path = 'course/{course_id}'.format(course_id=course_details.get('course'))
```

Comment:
this is the learner portal url for the course. We need to determine/finalize what we want when learner portal is off for enterprise

Reply:
resolved: we are not going to use the learner portal link in the 'start my course' link. it's only used for the My Dashboard link still for bulk enrollment use case